project/Build.scala

```diff
@@ -1331,16 +1331,32 @@ object Build {
         val inkuireLink = s"https://github.com/VirtusLab/Inkuire/releases/download/$inkuireVersion/inkuire.js"
         val inkuireDestinationFile = (Compile / resourceManaged).value / "dotty_res" / "scripts" / "inkuire.js"
         sbt.IO.touch(inkuireDestinationFile)
-        val downloadProcess = (new java.net.URL(inkuireLink) #> inkuireDestinationFile).run()
-        val result: Future[Int] = Future(blocking(downloadProcess.exitValue()))
-        val res = try {
-          Await.result(result, duration.Duration(60, "sec"))
-        } catch {
-          case _: TimeoutException =>
-            downloadProcess.destroy()
-            throw new MessageOnlyException(s"Failed to fetch inkuire.js from $inkuireLink: Download timeout")
+
+        def tryFetch(retries: Int, timeout: Int): Unit = {
+          val downloadProcess = (new java.net.URL(inkuireLink) #> inkuireDestinationFile).run()
+          val result: Future[Int] = Future(blocking(downloadProcess.exitValue()))
+          try {
+            Await.result(result, duration.Duration(timeout, "sec")) match {
+              case 0 =>
+              case res if retries > 0 =>
+                println(s"Failed to fetch inkuire.js from $inkuireLink: Error code $res. $retries retries left")
+                tryFetch(retries - 1, timeout)
+              case res => throw new MessageOnlyException(s"Failed to fetch inkuire.js from $inkuireLink: Error code $res")
+            }
+          } catch {
+            case e: TimeoutException =>
+              if(retries > 0) {
+                println(s"Failed to fetch inkuire.js from $inkuireLink: Download timeout. $retries retries left")
+                tryFetch(retries - 1, timeout)
+              }
+              else {
+                downloadProcess.destroy()
```

Comment:
Do we want also want this for the `if` branch above when we catch TimeoutException?

Reply:
yes, we should destroy process in both branches